src/react/components/pages/connections/connectionsPage.tsx

```diff
@@ -120,6 +121,10 @@ export default class ConnectionPage extends React.Component<IConnectionPageProps
     }
 
     private onFormSubmit = async (connection: IConnection) => {
+        const assetProvider = AssetProviderFactory.createFromConnection(connection);
+        if (assetProvider.initialize) {
```

Comment:
I like the idea of an initialize, but I feel it should be pushed back down into the factory itself vs. checking here if the assetProvider contains an initialization method.

Reply:
Sounds good. Just in the create?

Reply:
Thought we were removing this from here since it's in the save method now?